src/leveled_tictac.erl

```diff
@@ -241,21 +258,28 @@ alter_segment(Segment, Hash, Tree) ->
 %% Returns a list of segment IDs which hold differences between the state
 %% represented by the two trees.
 find_dirtyleaves(SrcTree, SnkTree) ->
-    Size = SrcTree#tictactree.size,
-    Size = SnkTree#tictactree.size,
+    SizeSrc = SrcTree#tictactree.size,
+    SizeSnk = SnkTree#tictactree.size,
+    true = SizeSrc == SizeSnk, 
     
     IdxList = find_dirtysegments(fetch_root(SrcTree), fetch_root(SnkTree)),
     SrcLeaves = fetch_leaves(SrcTree, IdxList),
     SnkLeaves = fetch_leaves(SnkTree, IdxList),
     
     FoldFun =
         fun(Idx, Acc) ->
-            {Idx, SrcLeaf} = lists:keyfind(Idx, 1, SrcLeaves),
-            {Idx, SnkLeaf} = lists:keyfind(Idx, 1, SnkLeaves),
+            SrcLeaf = element(2, lists:keyfind(Idx, 1, SrcLeaves)),
+            SnkLeaf = element(2, lists:keyfind(Idx, 1, SnkLeaves)),
             L2IdxList = segmentcompare(SrcLeaf, SnkLeaf),
-            Acc ++ lists:map(fun(X) -> X + Idx * ?L2_CHUNKSIZE end, L2IdxList)
+            lists:foldl(
+                fun(X, InnerAcc) ->
+                    SegID = X + Idx * ?L2_CHUNKSIZE,
+                    [SegID|InnerAcc]
+                end,
+                Acc,
+                L2IdxList)
         end,
-    lists:sort(lists:foldl(FoldFun, [], IdxList)).
+    lists:foldl(FoldFun, [], IdxList).
```

Comment:
You drop the fact that the list is sorted... on purpose?

Reply:
Ah... there are tests below that seem to show that indeed the list is always sorted